nx/lib/nx/backend.ex

```diff
@@ -234,10 +249,85 @@ defmodule Nx.Backend do
     chunk_each(dim - 1, rest, [doc | acc], limit, fun)
   end
 
-  defp inspect_value(%Complex{} = val), do: Complex.to_string(val)
-  defp inspect_value(integer) when is_integer(integer), do: Integer.to_string(integer)
-  defp inspect_value(float) when is_float(float), do: Float.to_string(float)
-  defp inspect_value(:neg_infinity), do: "-Inf"
-  defp inspect_value(:infinity), do: "Inf"
-  defp inspect_value(:nan), do: "NaN"
+  defp inspect_value(float, precision) when is_float(float) and is_integer(precision) do
+    float_to_string(float, precision)
+  end
+
+  defp inspect_value(float, nil) when is_float(float), do: Float.to_string(float)
+
+  defp inspect_value(%Complex{} = val, precision), do: complex_to_string(val, precision)
+
+  defp inspect_value(integer, _) when is_integer(integer), do: Integer.to_string(integer)
+  defp inspect_value(:neg_infinity, _), do: "-Inf"
+  defp inspect_value(:infinity, _), do: "Inf"
+  defp inspect_value(:nan, _), do: "NaN"
+
+  defp float_to_string(float, precision) do
+    [integer_part, decimal_part, exponent_part] =
+      case String.split(Float.to_string(float), [".", "e"], parts: 3) do
+        [i, d] -> [i, d, ""]
+        [i, d, e] -> [i, d, "e" <> e]
+      end
+
+    # We'll now prune decimal_part to ensure we have at most `precision`
+    # digits there.
+
+    decimal_part =
+      decimal_part
+      |> binary_part(0, min(byte_size(decimal_part), precision))
+
+    #  We also prune trailing zeros. Only for more than 1 digit because that single
+    # digit always needs to stay put.
+    decimal_part =
+      if byte_size(decimal_part) > 1 do
+        String.trim_trailing(decimal_part, "0")
+      else
+        decimal_part
+      end
+
+    integer_part <> "." <> decimal_part <> exponent_part
+  end
+
+  def complex_to_string(%Complex{re: re, im: im}, precision) do
+    re_str = complex_to_string_real(re, precision)
+    im_str = complex_to_string_component(im, precision)
+
+    "#{re_str}#{im_str}i"
+  end
+
+  defp complex_to_string_real(n, precision) do
+    case complex_to_string_component(n, precision) do
+      "-" <> _ = s -> s
+      "+" <> s -> s
+    end
+  end
+
+  defp complex_to_string_component(:infinity, _), do: "+Inf"
+  defp complex_to_string_component(:neg_infinity, _), do: "-Inf"
+  defp complex_to_string_component(:nan, _), do: "+NaN"
+
+  defp complex_to_string_component(n, nil) when n == 0, do: "+0.0"
+
+  defp complex_to_string_component(n, nil) do
+    abs_s = Kernel.to_string(abs(n))
+
+    if n >= 0 do
+      "+#{abs_s}"
+    else
+      "-#{abs_s}"
+    end
+  end
+
+  defp complex_to_string_component(n, precision) do
+    # convert number to float
+    n = if is_float(n), do: n, else: n * 1.0
+
+    str = float_to_string(n, precision)
```

Comment:
```suggestion
    str = float_to_string(n * 1.0, precision)
```

Reply:
This code means that, if precision is enabled, `1+2i` will always be printed as `1.0+2.0i`.